packages/react-provider/src/Provider.tsx

```diff
@@ -0,0 +1,71 @@
+import { useMergedRefs } from '@fluentui/react-hooks';
+import { getSlots, makeMergeProps } from '@fluentui/react-utils';
+import * as React from 'react';
+
+import { ThemeContext, ProviderContextValue, useFluent } from './context';
+import { PartialTheme } from '@fluentui/react-theme';
+import { ThemeProviderState, useThemeProviderState } from './ThemeProvider';
+
+export interface ProviderProps extends Partial<ProviderContextValue> {
+  theme?: PartialTheme;
+}
+export interface ProviderState extends ProviderContextValue, ThemeProviderState {}
+
+const mergeProps = makeMergeProps<ProviderState>();
+
+export function useProviderState(draftState: ProviderState) {
+  const parentContext = useFluent();
+
+  useThemeProviderState(draftState);
+
+  // TODO: add merge functions
+  draftState.document = draftState.document || parentContext.document;
+  draftState.dir = draftState.dir || parentContext.dir;
+}
+
+export function renderProvider(state: ProviderState) {
+  const { slots, slotProps } = getSlots(state);
+  const { theme } = state;
+
+  return (
+    <ThemeContext.Provider value={theme}>
```

Comment:
Oops, it should also render `ProviderContext`

Reply:
Shouldn't we use `ThemeProvider` here for `theme` and `WindowProvider` here for `target`?

Reply:
https://github.com/microsoft/fluentui/pull/16708#discussion_r567059203